src/client_side.cc

```diff
@@ -879,7 +879,9 @@ ClientSocketContextPushDeferredIfNeeded(Http::StreamPointer deferredRequest, Con
         /** NO data is allowed to have been sent. */
         assert(deferredRequest->http->out.size == 0);
         /** defer now. */
-        clientSocketRecipient(deferredRequest->deferredparams.node,
+        const auto savedNode = deferredRequest->deferredparams.node;
+        deferredRequest->deferredparams.node = nullptr;
```

Comment:
Why do we want to clear just `node` and not the whole `deferredRequest` structure, especially since #2086 appears to clear `deferredRequest.rep` member as well? If we should clear everything, let's do that in this PR. If we should not clear everything, let's leave a source code comment to explain why we should not.